### PR TITLE
Add TPC-H query 4 to TpchQueryBuilder

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

**Comment by majetideepak** [comment]
CC: @pramodsatya

File: velox/exec/tests/utils/TpchQueryBuilder.cpp [hunk 1]
```diff
@@ -291,6 +293,69 @@ TpchPlan TpchQueryBuilder::getQ3Plan() const {
...
+  std::vector<std::string> ordersColumns = {
+      "o_orderdate", "o_orderkey", "o_orderpriority"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+
+  // dateVal :2 = '1993-07-01'
+  const auto orderDate = "o_orderdate";
```
Comment (pramodsatya) [inline velox/exec/tests/utils/TpchQueryBuilder.cpp:311]: nit: the comment and this variable can be removed

File: velox/exec/tests/utils/TpchQueryBuilder.cpp [hunk 2]
```diff
@@ -291,6 +293,69 @@ TpchPlan TpchQueryBuilder::getQ3Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ4Plan() const {
+  static const std::string kOrders = "orders";
```
Comment (pramodsatya) [inline velox/exec/tests/utils/TpchQueryBuilder.cpp:297]: These variables are already defined in the header and can be removed

File: velox/exec/tests/utils/TpchQueryBuilder.cpp [hunk 3]
```diff
@@ -291,6 +293,69 @@ TpchPlan TpchQueryBuilder::getQ3Plan() const {
...
+  core::PlanNodeId lineitemPlanNodeId;
+  core::PlanNodeId ordersPlanNodeId;
+
+  auto orders = PlanBuilder(planNodeIdGenerator)
+                    .tableScan(
+                        kOrders,
+                        ordersSelectedRowType,
+                        ordersFileColumns,
+                        {orderDateFilter},
+                        "")
```
Comment (pramodsatya) [inline velox/exec/tests/utils/TpchQueryBuilder.cpp:325]: The tablescan can be simplified by removing the remaining filter

**Pull request closed without merging**